root_VS2012/programs/C#/Frameworks/Infrastructure/Public/Db/DamSqlDbWithMultiShard/DamSqlDbWithMultiShard/DamSqlDbWithMultiShard.cs

```diff
@@ -0,0 +1,797 @@
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        /// <summary>パラメタライズドクエリにパラメタを設定する。</summary>
+        /// <param name="parameterName">パラメタライズドクエリのパラメタ名</param>
+        /// <param name="obj">パラメタの値</param>
+        /// <remarks>通常、Ｄａｏ経由で利用する。</remarks>
+        public override void SetParameter(string parameterName, object obj)
+        {
+            this.SetParameter(parameterName, obj, null, -1, ParameterDirection.Input);
+        }
+
+        /// <summary>パラメタライズドクエリにパラメタを設定する。</summary>
+        /// <param name="parameterName">パラメタライズドクエリのパラメタ名</param>
+        /// <param name="obj">パラメタの値</param>
+        /// <param name="dbTypeInfo">パラメタの型（SqlDbType）（設定しない場合は、nullを指定）</param>
+        /// <remarks>通常、Ｄａｏ経由で利用する。</remarks>
+        public override void SetParameter(string parameterName, object obj, object dbTypeInfo)
+        {
+            this.SetParameter(parameterName, obj, dbTypeInfo, -1, ParameterDirection.Input);
+        }
+
+        /// <summary>パラメタライズドクエリにパラメタを設定する。</summary>
+        /// <param name="parameterName">パラメタライズドクエリのパラメタ名</param>
+        /// <param name="obj">パラメタの値</param>
+        /// <param name="dbTypeInfo">パラメタの型（SqlDbType）（設定しない場合は、nullを指定）</param>
+        /// <param name="size">パラメタのサイズ（設定しない場合は、-1を指定）</param>
+        /// <remarks>通常、Ｄａｏ経由で利用する。</remarks>
+        public override void SetParameter(string parameterName, object obj, object dbTypeInfo, int size)
+        {
+            this.SetParameter(parameterName, obj, dbTypeInfo, size, ParameterDirection.Input);
+        }
+
+        /// <summary>パラメタライズドクエリにパラメタを設定する。</summary>
+        /// <param name="parameterName">パラメタライズドクエリのパラメタ名</param>
+        /// <param name="obj">パラメタの値</param>
+        /// <param name="dbTypeInfo">パラメタの型（SqlDbType）（設定しない場合は、nullを指定）</param>
+        /// <param name="size">パラメタのサイズ（設定しない場合は、-1を指定）</param>
+        /// <param name="paramDirection">パラメタの方向</param>
+        /// <remarks>通常、Ｄａｏ経由で利用する。</remarks>
+        public override void SetParameter(string parameterName, object obj,
+            object dbTypeInfo, int size, ParameterDirection paramDirection)
+        {
+            if (this.IsDPQ)
+            {
+                // 動的パラメタライズド クエリ
+
+                // パラメタを保存（実行時に解析してまとめて設定する）
+                this._parameter.Add(parameterName, obj);
+                this._parameterType.Add(parameterName, dbTypeInfo);
+                this._parameterSize.Add(parameterName, size);
+                this._parameterDirection.Add(parameterName, paramDirection);
+            }
+            else
+            {
+                // 通常のパラメタライズド クエリ
+
+                // パラメタ・ライズド・クエリにパラメタを設定する。
+                SqlParameter param = new SqlParameter();
+
+                #region dbTypeInfo
+
+                // nullの場合は、設定しない。
+                if (dbTypeInfo != null)
+                {
+                    // フラグ
```

Comment:
@daisukenishino 

Please describe the detail or meaning of this flag.
So, this program will be easy to read.